pkgs/misc/vscode-extensions/vscode-utils.nix

```diff
@@ -28,12 +24,12 @@ let
   }:
   stdenv.mkDerivation ((removeAttrs a [ "vscodeExtUniqueId" ]) //  {
 
-    name = namePrefix + name;
+    name = "vscode-extension-${name}";
 
     inherit vscodeExtUniqueId;
     inherit configurePhase buildPhase dontPatchELF dontStrip;
 
-    installPrefix = "share/${extendedPkgName}/extensions/${vscodeExtUniqueId}";
+    installPrefix = "${vscodeExtUniqueId}";
```

Comment:
What bothers me is that now, an extension no longer stand as a standard package which could simply be installed in an environment using nix-env. It this really a use case, I'm not sure.

In case we want to treat those extensions as standalone packages, stuff like this really should end up under the lfs share folder and the namespace under share being fairly unique. It was the approach taken by *vim plugins* which I initially took inspiration from.

It might even end up being required by *licenses* to expose the source code / files when installing `vscode-with-extensions`/`vscodium-with-extensions`/etc (is it?).

How about instead just outputting those under something more straigtforward such as: `share/vscode-extensions` regardless of the wrapped executable?

Note that this is **not a strong opinion**. Anyone else interested in providing more insight?

Reply:
That's a very good point! However that isn't a perfect solution, since Insiders versions of VSCode need their extensions to be in an appropriately named directory.
I wonder how many people actually use Nix-managed extensions but do not use them via vscode-with-extensions, but I agree that it's good to support it on principle.
My suggestion is that we have the extension derivations like they are now, and provide versions of them that have been symlinked to the appropriate place that people can use. This means the path could be varied depending on whether they use an Insiders version or not.

Reply:
> Insiders versions of VSCode need their extensions to be in an appropriately named directory.

I'm not sure to quite understand in what regards the current PR changes that and what exactly is the distinction in terms of extension location?